dlib/dnn/loss.h

```diff
@@ -1035,14 +1094,19 @@ namespace dlib
                 throw impossible_labeling_error(sout.str());
             }
 
-            det_idx = find_best_detection_window(rect,label); 
+            det_idx = find_best_detection_window(rect,label,scale_invariant);
+            const double scale = scale_invariant
```

Comment:
Use an if/else.  This bit of code looks too crazy otherwise.  I thought it was just a tabbing error at first.

Reply:
Sorry. :)

I guess I just wanted to keep the constness.

Fixed.